vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html

```diff
@@ -206,7 +222,102 @@
             		$('#imageId').prop('selectedIndex',0).trigger( "change" );
             	}
             }
-            
+			
+			function embedVideo() {
+				var videoID = $("#saveVideo").data('value'); // gets video ID
+				var oldUrl = $("#saveVideo").data('url');
+				//var oldTitle = $("#VideoTitle").data('title');
+				var url = $("#videoLink").val();
+				var title = $("#videoTitle").val();
+				var vidFile = $("#videoFile")[0].files[0];
+				var reader = new FileReader();
+				var videoBlock = "";
+				var ajaxUrl = '';
+				var oldVideoBlock = '';
+				var formData = new FormData();
+				if (videoID) {
+					oldVideoBlock = $("#" + videoID);
+					var videoBlockId = videoID;
+					ajaxUrl = "[(@{'/staff/module/'+${module.id}+'/slide/'+${slide.id}+'/video/'})]" + videoBlockId + '?[(${_csrf.parameterName})]=[(${_csrf.token})]';
+					formData.append('videoBlockId', videoBlockId);
+
+					if (url) {
+						formData.append('url', url);
+						videoBlock = createVideoBlock(reader, url, title);
+						$("#" + videoBlockId).replaceWith(videoBlock);
+					}
+					if (vidFile) {
+						formData.append('videoFile', vidFile);
+						reader.onload = function (theFile) {
+							videoBlock = createVideoBlock(reader, theFile.target.result);
+							$("#" + videoBlockId).replaceWith(videoBlock);
+						};
+					}
+				}
+				else {
+					if (url) {
+						formData.append('url', url);
+						videoBlock = createVideoBlock(reader, url, title);
+						$('#slideSpace').append(videoBlock);
+					}
+					if (vidFile) {
+						formData.append('videoFile', vidFile);
+						reader.onload = function (theFile) {
+							videoBlock = createVideoBlock(reader, theFile.target.result, title);
+							$('#slideSpace').append(videoBlock);
+						}
+					}
+					ajaxUrl = "[(@{'/staff/module/'+${module.id}+'/slide/'+${slide.id}+'/video?'+${_csrf.parameterName}+'='+${_csrf.token}})]";
+				}
```

Comment:
there seems to be a lot of duplicate code in the above if/else statement. That should be more DRY.